roundkornerlayout/src/main/java/com/jcminarro/roundkornerlayout/Extensions.kt

```diff
@@ -1,10 +1,87 @@
 package com.jcminarro.roundkornerlayout
 
+import android.content.res.TypedArray
+import android.graphics.Path
+import android.graphics.RectF
 import android.os.Build
 import android.view.View
 
+internal fun View.updateOutlineProvider(cornersHolder: CornersHolder) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+        outlineProvider = RoundOutlineProvider(cornersHolder)
+    }
+}
+
 internal fun View.updateOutlineProvider(cornerRadius: Float) {
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
         outlineProvider = RoundOutlineProvider(cornerRadius)
     }
-}
\ No newline at end of file
+}
+
+internal fun Path.addRoundRectWithRoundCorners(rectF: RectF, cornersHolder: CornersHolder) {
+    addRoundRectWithRoundCorners(
+            rectF,
+            cornersHolder.topLeftCornerRadius,
+            cornersHolder.topRightCornerRadius,
+            cornersHolder.bottomRightCornerRadius,
+            cornersHolder.bottomLeftCornerRadius
+    )
+}
+
+internal fun Path.addRoundRectWithRoundCorners(rectF: RectF,
+                                               topLeftCornerRadius: Float,
+                                               topRightCornerRadius: Float,
+                                               bottomRightCornerRadius: Float,
+                                               bottomLeftCornerRadius: Float) {
+    addRoundRect(
+            rectF,
+            floatArrayOf(
+                    topLeftCornerRadius,
+                    topLeftCornerRadius,
+                    topRightCornerRadius,
+                    topRightCornerRadius,
+                    bottomRightCornerRadius,
+                    bottomRightCornerRadius,
+                    bottomLeftCornerRadius,
+                    bottomLeftCornerRadius
+            ),
+            Path.Direction.CW
+    )
+}
+
+internal fun TypedArray.getCornerRadius(attrCornerRadius: Int,
+                                        attrTopLeftCornerRadius: Int,
+                                        attrTopRightCornerRadius: Int,
+                                        attrBottomRightCornerRadius: Int,
+                                        attrBottomLeftCornerRadius: Int
+): CornersHolder {
+    val cornerRadius = getDimension(attrCornerRadius, 0f)
+    var topLeftCornerRadius = cornerRadius
+    var topRightCornerRadius = cornerRadius
+    var bottomRightCornerRadius = cornerRadius
+    var bottomLeftCornerRadius = cornerRadius
+
+    val _topLeftCornerRadius = getDimension(attrTopLeftCornerRadius, Float.NaN)
```

Comment:
Instead of use `Float.NaN` you could use the default value that is `cornerRadius`, and you don't need to check every value is different than `Float.NaN` to perform the assignation